Set policies through version 3.9 to allow better IPO support (#57)
minetest/irrlicht: CMakeLists.txt

cmake_minimum_required(VERSION 3.5)

# Policies should be set to NEW behavior whenever possible. This only sets
# policies up to version 3.9 (for the IPO policies we want) as a way to
# introduce the feature more gradually and see how it works.
if(${CMAKE_VERSION} VERSION_LESS 3.9)
	cmake_policy(VERSION ${CMAKE_VERSION_MAJOR}.${CMAKE_VERSION_MINOR})
else()
	cmake_policy(VERSION 3.9)

> endif()

Reviewer: Wouldn't `cmake_policy(VERSION 3.9)` just work? I don't see anything here implying the version can't be higher than the current version:
https://cmake.org/cmake/help/v3.5/command/cmake_policy.html

Author: If I understood correctly policies cannot be set to NEW on versions before the policy was introduced.

Reviewer: Of course not, but with cmake_policy 3.9 you're just saying to set all policies before/on 3.9 to NEW.
Now CMake 3.7 wouldn't know about policies in 3.8 or 3.9 (the future) but that doesn't mean it can't work.

Author: I feel like there'd be a problem but I don't know what. I took the code mostly from https://cliutils.gitlab.io/modern-cmake/chapters/basics.html and I've always understood that you want to set the policies for the version of CMake you are currently running. I would test your suggestion but my CMake version is too recent.

Author: https://gitlab.kitware.com/cmake/cmake/-/issues/20392

Reviewer: Hm, I see.

cmake_minimum_required(VERSION 3.5)

> # Policies should be set to NEW behavior whenever possible. This only sets
> # policies up to version 3.9 (for the IPO policies we want) as a way to
> # introduce the feature more gradually and see how it works.

Reviewer: ```suggestion
# Set policies up to 3.9 since we want to enable the IPO option
```
